libcudacxx/include/cuda/std/__cccl/diagnostic.h

```diff
@@ -127,7 +127,7 @@
 #    define _CCCL_NV_DIAG_SUPPRESS(_WARNING) _CCCL_PRAGMA(diagnostic push) _CCCL_PRAGMA(diag_suppress _WARNING)
 #    define _CCCL_NV_DIAG_DEFAULT(_WARNING)  _CCCL_PRAGMA(diagnostic pop)
 #  else // ^^^ __NVCC_DIAG_PRAGMA_SUPPORT__ ^^^ / vvv !__NVCC_DIAG_PRAGMA_SUPPORT__ vvv
-#    if _CCCL_COMPILER(MSVC2017) // MSVC 2017 has issues with restoring the warning
+#    if _CCCL_COMPILER(MSVC2017) || _CCCL_COMPILER(GCC) // these compilers have issues with restoring the warning
```

Comment:
This is not valid, this should be `_CUDA_COMPILER_CUDACC_BELOW(11, 5)`

We want to not check against GCC but against the cuda version

Reply:
`__NVCC_DIAG_PRAGMA_SUPPORT__` was introduced with CTK 11.5. From experimental evaluation, it seems that this is the case we want to fix. The line I changed is in the branch of  `elif defined(__NVCC_DIAG_PRAGMA_SUPPORT__) || _CCCL_COMPILER(ICC)`.

Reply:
Is that indeed for all gcc compiler or only some?

Reply:
Double-checked: it applies to all gcc versions.